contrib/babelfishpg_tsql/src/pltsql.h

```diff
@@ -2348,4 +2348,29 @@ extern bool validate_special_function(char *proc_nsname, char *proc_name, int na
  */
 extern char *tsql_format_type_extended(Oid type_oid, int32 typemod, bits16 flags);
 
+/* Restricted objects definition */
+typedef struct
+{
+	const char *name;
+	ObjectType type;
+} RestrictedObject;
+
+/* Separate arrays for each schema */
+static const RestrictedObject master_dbo_objects[] = {
+	{"xp_qv",					OBJECT_PROCEDURE},
+	{"xp_instance_regread",		OBJECT_PROCEDURE},
+	{"sp_addlinkedserver",		OBJECT_PROCEDURE},
+	{"sp_addlinkedsrvlogin",	OBJECT_PROCEDURE},
+	{"sp_droplinkedsrvlogin",	OBJECT_PROCEDURE},
+	{"sp_dropserver",			OBJECT_PROCEDURE},
+	{"sp_enum_oledb_providers",	OBJECT_PROCEDURE},
+	{"sp_testlinkedserver",		OBJECT_PROCEDURE}
+};
+
+static const RestrictedObject msdb_dbo_objects[] = {
```

Comment:
I would suggest instead of having list for each schemas, have something which can be extendend to other schemas if required.

Reply:
I considered a hash table but stuck with arrays since we’ve got a small number of restricted objects. It keeps things simple and performs well. If things change, we can always tweak it later.